### ISSUE TITLE
When publishing from VS2017 it uses the project file name instead of assembly name.

### ISSUE THREAD

**Issue by johnwc** [aspnet/MvcPrecompilation#125]
As you can see below, it is using the file name of the csproj instead of the assembly name. You can see the error says `Could not load file or assembly 'My Web Portal, ...` If I rename the project name to the same name as the assembly, it works. The name I have for the assembly under the properties for the project is `My.Web.Portal`

```
Bundler: Begin processing bundleconfig.json
Bundler: Done processing bundleconfig.json
D:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1964,5): Warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
C:\Program Files (x86)\Microsoft SDKs\TypeScript\2.1\tsc.exe  --noEmitOnError --listEmittedFiles --locale en-US
C:\Program Files (x86)\Microsoft SDKs\TypeScript\2.1\tsc.exe  --noEmitOnError --listEmittedFiles --locale en-US
D:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\Roslyn\csc.exe /noconfig /unsafe- /checked- /nowarn:1701,1702,1705,2008 /nostdlib+ /errorreport:prompt /warn:4 /define:TRACE;RELEASE;NETCOREAPP1_1 /errorendlocation /preferreduilang:en-US /reference:... /debug- /debug:portable /filealign:512 /nologo /optimize+ /out:obj\Release\netcoreapp1.1\My.Web.Portal.dll /ruleset:"D:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Team Tools\Static Analysis Tools\\Rule Sets\MinimumRecommendedRules.ruleset" /target:exe /warnaserror- /utf8output /deterministic+ /analyzer:Microsoft.CodeAnalysis.Analyzers.dll /analyzer:Microsoft.CodeAnalysis.CSharp.Analyzers.dll ... "...\Temp\.NETCoreApp,Version=v1.1.AssemblyAttributes.cs" ...
My.Web.Portal -> ...\Source\Web\Web Portal .Net Core\bin\Release\netcoreapp1.1\My.Web.Portal.dll
dotnet exec --runtimeconfig "...\Source\Web\Web Portal .Net Core\bin\Release\netcoreapp1.1\My.Web.Portal.runtimeconfig.json" --depsfile "...\Source\Web\Web Portal .Net Core\bin\Release\netcoreapp1.1\My.Web.Portal.deps.json" "...\.nuget\packages\microsoft.aspnetcore.mvc.razor.viewcompilation\1.1.1-rtm-201704210746\build\netcoreapp1.1\../../lib/netcoreapp1.1/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.dll" @"obj\Release\netcoreapp1.1\microsoft.aspnetcore.mvc.razor.viewcompilation.rsp"
Could not load file or assembly 'My Web Portal, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks, IntPtr ptrLoadContextBinder)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks, IntPtr ptrLoadContextBinder)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.MvcServiceProvider.GetConfigureCompilationAction(String configureCompilationType)
   at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.MvcServiceProvider..ctor(String projectPath, String applicationName, String contentRoot, String configureCompilationType)
   at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.PrecompileRunCommand.Execute()
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.PrecompilationApplication.Execute(String[] args)
...\.nuget\packages\microsoft.aspnetcore.mvc.razor.viewcompilation\1.1.1-rtm-201704210746\build\netcoreapp1.1\Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets(25,5): Error MSB3073: The command "dotnet exec --runtimeconfig "...\Source\Web\Web Portal .Net Core\bin\Release\netcoreapp1.1\My.Web.Portal.runtimeconfig.json" --depsfile "...\Source\Web\Web Portal .Net Core\bin\Release\netcoreapp1.1\My.Web.Portal.deps.json" "...\.nuget\packages\microsoft.aspnetcore.mvc.razor.viewcompilation\1.1.1-rtm-201704210746\build\netcoreapp1.1\../../lib/netcoreapp1.1/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.dll" @"obj\Release\netcoreapp1.1\microsoft.aspnetcore.mvc.razor.viewcompilation.rsp"" exited with code 1.

2>Publish failed due to build errors. Check the error list for more details.
========== Build: 1 succeeded or up-to-date, 0 failed, 0 skipped ==========
========== Publish: 0 succeeded, 1 failed, 0 skipped ==========
```

**Comment by pranavkm** [comment]
The issue was fixed as part of https://github.com/aspnet/MvcPrecompilation/pull/82. @Eilon should we backport the fix to 1.1.x?

**Comment by johnwc** [comment]
That issue looks to be related to the name of the output, not the referenced input. Am I reading that wrong?

**Comment by pranavkm** [comment]
@johnwc - the tool was incorrectly using `$(MSBuildProjectName).dll` instead of `$(AssemblyName)` when passing in arguments to the view compilation tool.

**Comment by johnwc** [comment]
The diff in that closed issue only showed a output property being updated. Was the input also updated?

**Comment by Eilon** [comment]
@ajaybhargavb can you take a look?

**Comment by ajaybhargavb** [comment]
@johnwc, [here](https://github.com/aspnet/MvcPrecompilation/pull/82/files#diff-de06dd599c04ef47f24c51b91a65819fR24). We changed the `ExecArgs` property to use the assembly name. Are you seeing this issue even after upgrading to 2.0.0-preview1?

**Comment by mwijnands** [comment]
Backporting the fix to 1.1.x would be nice. Can't use precompilation if my assembly name is different from my project name.

**Comment by ajaybhargavb** [comment]
Filed a new issue for this https://github.com/aspnet/MvcPrecompilation/issues/133.
